src/main/java/domain/LadderResult.java

```diff
@@ -0,0 +1,39 @@
+package domain;
+
+import java.util.Map;
+
+public class LadderResult {
+    public static final String PLAYER_NOT_FOUND = "해당 Player가 존재하지 않습니다.";
+
+    private final Map<Player, String> ladderResult;
+
+    public LadderResult(Map<Player, String> ladderResult) {
+        this.ladderResult = ladderResult;
+    }
+
+    public String getResult(Player player) {
+        if (!ladderResult.containsKey(player)) {
+            throw new IllegalArgumentException(PLAYER_NOT_FOUND);
+        }
+        return ladderResult.get(player);
+    }
+
+    public String getResult(String playerName) {
+        Player player = Player.from(playerName);
+        if (!ladderResult.containsKey(player)) {
+            throw new IllegalArgumentException(PLAYER_NOT_FOUND);
+        }
+        return ladderResult.get(player);
+    }
```

Comment:
```suggestion
    public String getResult(String playerName) {
        return getResult(Player.from(playerName);
    }
```

같은 로직이 중복될 필요는 없어요.

분은 아닙니다만, 예외 발생 시 프로그램이 바로 종료될 수 있기 때문에 여기서 종료를 꼭 해야만 하는 경우가 아니라면 예외 사용은 조금 더 신중하게 결정해도 됩니다.
입력한 사용자 이름이 잘못된다면 오류라고 판단하는게 나을까요? 아니면 "참여자가 아닙니다" 같은 메시지를 뿌려주는게 더 나을까요?

정답은 정하기 나름이지만 제 생각에는 굳이 프로그램을 끝낼 요소는 아니라고 보여 덧붙여보았습니다.

Reply:
점진적 리팩토링을 해보려다가 동일한 로직이 겹쳤네요.. 해당 예외가 발생한다고 프로그램이 종료되지느 않고, 말씀하신대로 메시지를 남기고 다시 입력받습니다.